### PR TITLE
Fix / EmojiCompat not initialized

### LINKED ISSUE #530: Crash / EmojiCompat not initialized
https://github.com/matrix-org/riot-android-rageshakes/issues/6818

Thread: main, Exception: java.lang.IllegalStateException: Not initialized yet
at androidx.core.util.Preconditions.checkState(Preconditions.java:96)
at androidx.emoji.text.EmojiCompat.process(EmojiCompat.java:742)
at androidx.emoji.text.EmojiCompat.process(EmojiCompat.java:702)
at androidx.emoji.text.EmojiCompat.process(EmojiCompat.java:666)
at androidx.emoji.text.EmojiCompat.process(EmojiCompat.java:634)
at im.vector.riotx.features.reactions.widget.ReactionButton.setReactionString(ReactionButton.kt:81)
at

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #530

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/VectorApplication.kt [hunk 1]
```diff
@@ -109,21 +109,7 @@ class VectorApplication : Application(), HasVectorInjector, MatrixConfiguration.
...
-                .registerInitCallback(object : EmojiCompat.InitCallback() {
-            override fun onInitialized() {
-                Timber.v("Emoji compat onInitialized success ")
-            }
-
-            override fun onFailed(throwable: Throwable?) {
-                Timber.e(throwable,"Failed to init EmojiCompat")
-            }
-        })
+        EmojiCompatHelper.init(this)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/VectorApplication.kt]: pass fontRequest as a param?

File: vector/src/main/java/im/vector/riotx/EmojiCompatHelper.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import android.content.Context
+import androidx.core.provider.FontRequest
+import androidx.emoji.text.EmojiCompat
+import androidx.emoji.text.FontRequestEmojiCompatConfig
+import timber.log.Timber
+
+object EmojiCompatHelper {
+
+    var initialized = false
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/EmojiCompatHelper.kt]: private?

File: vector/src/main/java/im/vector/riotx/EmojiCompatHelper.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ */
+package im.vector.riotx
+
+import android.content.Context
+import androidx.core.provider.FontRequest
+import androidx.emoji.text.EmojiCompat
+import androidx.emoji.text.FontRequestEmojiCompatConfig
+import timber.log.Timber
+
+object EmojiCompatHelper {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/EmojiCompatHelper.kt]: Maybe call it `EmojiCompatWrapper`? And make it a singleton dagger style?

File: vector/src/main/java/im/vector/riotx/VectorApplication.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/VectorApplication.kt]: done

File: vector/src/main/java/im/vector/riotx/EmojiCompatHelper.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/EmojiCompatHelper.kt]: done

File: vector/src/main/java/im/vector/riotx/EmojiCompatHelper.kt (hunk 3 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/EmojiCompatHelper.kt]: done

**Pull request merged**